test/integration/open-with/OpenWith.int.js

```diff
@@ -0,0 +1,142 @@
+// <reference types="Cypress" />
+import l from '../../support/i18n';
+
+describe('OpenWith', () => {
+    beforeEach(() => {
+        cy.server();
+        cy.route('GET', '**/files/*?fields=extension', 'fixture:open-with/file-extension-document');
+    });
+
+    it('A single integration', () => {
+        cy.route('GET', '**/files/*/open_with_integrations', 'fixture:open-with/integration-adobe');
+        cy.visit('/ContentOpenWith');
+
+        // The button should be enabled
+        cy.getByTestId('singleintegrationbutton').should('not.have.attr', 'aria-disabled', 'true');
+
+        // Hover over the Open With button
+        cy.getByTestId('bcow-content').trigger('mouseover');
+        // Tooltip should render on mouseover
+        cy.get('[role="tooltip"]').contains('Send this document for signature');
+    });
+
+    it('A custom integration', () => {
+        cy.route('GET', '**/files/*/open_with_integrations', 'fixture:open-with/integration-custom');
+        cy.visit('/ContentOpenWith');
+
+        // The button should be enabled
+        cy.getByTestId('singleintegrationbutton').should('not.have.attr', 'aria-disabled', 'true');
+
+        // Hover over the Open With button
+        cy.getByTestId('bcow-content').trigger('mouseover');
+        // Tooltip should render on mouseover
+        cy.get('[role="tooltip"]').contains('This is a custom integration');
+    });
+
+    it('Multiple integrations', () => {
+        cy.route('GET', '**/files/*/open_with_integrations', 'fixture:open-with/integrations-multiple');
+        cy.visit('/ContentOpenWith');
+
+        // Click the Open With button
+        cy.getByTestId('multipleintegrationsbutton').click();
+        cy.get('[role="menuitem"]').as('menuItem');
+
+        // The dropdown should render in sorted order
+        cy.get('@menuItem')
+            .eq(0)
+            .contains('Open');
+        cy.get('@menuItem')
+            .eq(1)
+            .contains('Google Docs (beta)');
+        cy.get('@menuItem')
+            .eq(2)
+            .contains('Adobe Sign');
+        cy.get('@menuItem')
+            .eq(3)
+            .contains('My custom integration');
+    });
+
+    describe('box edit', () => {
+        it('box edit is uninstalled', () => {
```

Comment:
Having `describe('box-edit')` and `it('box-edit')` will show some repetitive text in the test descriptions.